CI/appveyor.install.ps1

```diff
@@ -65,6 +114,61 @@ function CheckAndInstall([string] $dependencyName, [string] $signalFile, [script
 }
 
 # installation functions
+function InstallSevenZ() {
+  if($64Bit){
+    $downloadUrl = "http://www.7-zip.org/a/7z1701-x64.exe"
+  } else {
+    $downloadUrl = "http://www.7-zip.org/a/7z1701.exe"
+  }
+  DownloadFile "$downloadUrl" "7z-installer.exe"
+  Step "installing 7z"
+  exec ".\7z-installer.exe" ("/S", "/D=`"C:\Program Files\7-Zip`"")
+}
+
+function InstallCmake() {
+  DownloadFile "https://cmake.org/files/v3.9/cmake-3.9.4-win32-x86.msi" "cmake-installer.msi"
+  Step "installing cmake"
+  exec "msiexec.exe" @("/q", "/li", "$workingBaseDir\cmake-installer.log", "/i", "cmake-installer.msi")
+  if(Test-Path -Path "$workingBaseDir\cmake-installer.log" -PathType Leaf){
+    Get-Content "$workingBaseDir\cmake-installer.log" | Out-File $logFile -Append
+  }
+}
+
+function InstallMsys() {
+  DownloadFile "https://sourceforge.net/projects/mingwbuilds/files/external-binary-packages/msys%2B7za%2Bwget%2Bsvn%2Bgit%2Bmercurial%2Bcvs-rev13.7z/download" "msys.7z" $true
+  if (!(Test-Path -Path "C:\MinGW\msys\1.0" -PathType Container)) {
+    Step "Creating MinGW path"
+    New-Item -Path "C:\MinGW\msys\1.0" -ItemType "directory" >> "$logFile" 2>&1
+  }
+  ExtractZip "msys.7z" "."
+  Step "Copying folder"
+  Move-Item "msys\*" "C:\MinGW\msys\1.0" >> "$logFile" 2>&1
+}
+
+function InstallBoost() {
+  DownloadFile "https://sourceforge.net/projects/boost/files/boost/1.60.0/boost_1_60_0.tar.gz/download" "boost.tar.gz" $true
+  if (!(Test-Path -Path "C:\Libraries\" -PathType Container)) {
+    Step "Creating Boost path"
+    New-Item -Path "C:\Libraries\" -ItemType "directory" >> "$logFile" 2>&1
+  }
+  ExtractTar "boost.tar.gz" "."
+  Step "Copying folder"
+  Move-Item "boost_1_60_0" "C:\Libraries\" >> "$logFile" 2>&1
+}
+
+function InstallQt() {
+  DownloadFile "http://download.qt.io/official_releases/qt/5.6/5.6.3/qt-opensource-windows-x86-mingw492-5.6.3.exe" "qt-installer.exe" $true
```

Comment:
I've built with Qt 5.9.3 and 5.10.0 (with Qt Creator 4.5.0 which has a much nicer CMake interface BTW) - is this the highest version that can be got working on AppVeyor?

Reply:
Ah, AppVeyor's default build environment has:
* Qt 5.9.2: `C:\Qt\5.9.2` (`C:\Qt\5.9` mapped to `C:\Qt\5.9.1` for backward compatibility) with MinGW 5.3.0 32 bit: `C:\Qt\5.9.2\mingw53_32`
* Qt 5.8.0: `C:\Qt\5.8` with MinGW 5.3.0 32 bit: `C:\Qt\5.8\mingw53_32`
* Qt 5.7.1: `C:\Qt\5.7` with MinGW 5.3.0 32 bit: `C:\Qt\5.7\mingw53_32`
* Qt 5.6.2: `C:\Qt\5.6` with MinGW 4.9.2 32 bit: `C:\Qt\5.6\mingw49_32`
* Qt 5.5: `C:\Qt\5.5` with MinGW 4.9.2 32 bit: `C:\Qt\5.5\mingw492_32`
* Qt 5.4: `C:\Qt\5.4` with MinGW 4.9.1 (32 bit) OpenGL: `C:\Qt\5.4\mingw491_32`
* Qt 5.3: `C:\Qt\5.3` with MinGW 4.8.2 (32 bit): `C:\Qt\5.3\mingw482_32`

Reply:
We had issues with Qt 5.9 in the past, which is why we are not yet building against it.

Reply:
Anything in between 5.6.3 and 5.9.2 work for us (IIRC we need 5.7.1 for the QGamePad module though that does have an external dependency on older Windows 7 which we cannot bundle - the Jun10 DirectX SDK that provides `XInput1.3.dll`)

Reply:
It does not on Windows, see https://github.com/Mudlet/Mudlet/issues/1176 for background.

Reply:
Ah, looking at that issue I realise my builds would not have exercised that area of the code.  Trouble is I *think* any libraries that we use have to have been built with a set of options compatible with the mingw built Qt version - or is it just the main Mudlet application - I dunno this Windows stuff is not as easy as other OS; guess that is why I tried to do a Cygwin port; now that was more POSIX-y to my tastes but the Qt5 support was ropey back when I was considering it originally...

Reply:
Yeah - so this part is correct, it needs to stay like this.